lib/rack/utils.rb

```diff
@@ -176,8 +177,16 @@ def best_q_match(q_value_header, available_mimes)
       matches&.first
     end
 
-    # Escape ampersands, brackets and quotes to their HTML/XML entities.
-    define_method(:escape_html, CGI.method(:escapeHTML))
+    # Introduced in ERB 4.0. ERB::Escape is an alias for ERB::Utils which
+    # doesn't get monkey-patched by rails
+    if defined?(ERB::Escape) && ERB::Escape.instance_method(:html_escape)
```

Comment:
This rats nest of code to shave a % of runtime seems like a bad tradeoff. It makes the code less readable and maintainable.

Reply:
Sorry for the unkind characterization. "Convoluted" would be better.

Reply:
I don't particularly see the importance of using `define_method` for only 1% improvement but `ERB::Escape` turns out to be about 50% faster when the text doesn't need escaping. So I would say using `ERB::Escape` is nice, though I would prefer to write it with plain `def` myself.